File: backend/src/main/kotlin/com/plugga/backend/service/PileService.java
package com.plugga.backend.service;

import com.plugga.backend.entity.Pile;

import java.util.List;

public interface PileService {
    List<Pile> findAll();

    Pile findById(int id);

    void save(Pile pile);

    void deleteById(int id);
}

File: backend/src/main/kotlin/com/plugga/backend/service/PileService.kt
package com.plugga.backend.service

import com.plugga.backend.entity.Pile
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface PileService {
    fun findAll(pageable: Pageable): Page<Pile>

    fun findById(id: Int): Pile?

    fun findByDeckId(pageable: Pageable, id: Int): Page<Pile>

    fun save(pile: Pile)

    fun deleteById(id: Int)
}
